datatables.go

```diff
@@ -328,34 +324,48 @@ func (p *Parser) bindGrenadeProjectiles(event st.EntityCreatedEvent) {
 
 	// @micvbang: not quite sure what the difference between Thrower and Owner is.
 	event.Entity.FindProperty("m_hThrower").RegisterPropertyUpdateHandler(func(val st.PropValue) {
-		throwerID := val.IntVal & indexMask
-		throwerIndex := throwerID - 1
-
-		thrower := p.entityIDToPlayers[throwerIndex]
-		proj.Thrower = thrower
+		proj.Thrower = p.propValueToPlayer(val)
+
+		// The Owner of a grenade is not modified in CWeaponCSBase when the grenade is thrown. In order to keep
+		// Player.CurrentEquipment updated, we remove thrown grenades here.
+		// @micvbang FYI: once a grenade is thrown it becomes a grenade projectile. The entity ID of a grenade
+		// projectile is _not_ the same as the grenade that was thrown. I haven't been able to find a better
+		// way to identify the thrown grenade, other than simply looping over all CurrentEquipment.
+		for _, eq := range proj.Thrower.CurrentEquipment {
+			if eq.Weapon == proj.Weapon {
+				ammo := proj.Thrower.AmmoLeft[eq.AmmoType]
```

Comment:
This is a problem for `cs-demos/valve_matchmaking.dem`

https://travis-ci.org/markus-wa/demoinfocs-golang/jobs/400343552#L612

Reply:
Ah. Without going deeper into it, I would bet that it has to do with the thrower not being set for fire grenades in some demos.